designdocs/batched_messages.md

```diff
@@ -10,98 +10,198 @@
+  [Avro](https://avro.apache.org/docs/current/) are supported.
+* `PayloadLocation` - The location of the payload. Can be either `OnChain` or `OffChain`.
+  * `OnChain`
+  * `OffChain`
+* `Payload`
+  * `OnChain`
+    * `source`: `MsaId`
+    * `payload`: `Vec<u8>`
+  * `IPFS`
+    * `payload_cid`: `TBD`
+    * `payload_byte_length`: `TBD`
+
+### Traits
+* `Model` - TBD. A common interface for accessing message payload information.
+  * Derives `Encode`, `Decode`, `MaxEncodedLen`
+  * `max_length`: `SchemaMaxBytesBoundedVecLimit`
 
 ### Types
-* `BatchAnnouncementParams<T:Config>`: generic
-    * `batch_uri`:`Vec<u8>` the URI of the batch file. Must be an IPFS [CIDv1](https://github.com/multiformats/cid/) URI. Accepted codec, algorithm, and base are to be determined.
-    * `message_schema_id`: `SchemaId`  the schema id for the messages in this batch. The `schema_id` must refer to schemas used for batching only.
-    * `file_size`: `usize`, the size of the batch file, used to determine message fee as well as to let consumers know what size files to expect.  Must be &gt;= the minimum possible DSNP batch file size.
-    * `file_format`: `BatchFormat`, indicator of the file format of the batch file.
-
-The file hash is not included separately. Since the `batch_uri` uses CIDv1 specification, the file hash is already included.
+* `Schema<T:Config>`: generic
+    * `model_type`: `ModelType` See enum section above.
+    * `model`: `Model` Defines the shape of the message payload.
+    * `payload_location`: `PayloadLocation` See enum section above.
 
-* `BatchAnnouncement`: implements `BatchAnnouncementParams`, returned by `get_batches`
+* `Message<T:Config>`: generic
+    * `schema_id`: `u16`
+    * `source`: `MsaId` Source of the message.
+    * `provider`: `MsaId` Public key of a capacity-providing account
+    * `payload`: `Payload` The payload.
 
-See the [implementation of paging in the messages pallet](https://github.com/LibertyDSNP/mrc/blob/main/common/primitives/src/messages.rs#L26-L58) for comparison.
-
-* `BatchAnnouncementOptions<T:Config>`
-    * `msa_id`:  `Option<MsaId>`, the announcer's MSA id.  Pass None() to get all announcements.
-    * `from_block`: `<T::BlockNumber>`, retrieve messages starting at the given block number (inclusive)
-    * `to_block`: `<T::BlockNumber>`, retrieve messages ending at the given block number (inclusive)
-    * `from_index`: `u32`, starting message index
-    * `page_size`: `usize`, retrieve `page_size` messages at a time, up to configured `T::BatchPageSizeMax`. If 0, return `T::BatchPageSizeMax` results
-
-* `BatchAnnouncementResult`
-    * `has_next`: `uint32`, current page number
-    * `next_block`: `<T::BlockNumber>` starting block number of next page of results
-    * `next_index`: `u32` starting index of next results in `next_block`
-    * `results`: `Vec<BatchAnnouncement>`
+(See alternatives section for another way to structure payloads)
```

Comment:
I don't actually understand this way of structuring it. The alt section I do, but I don't understand why 
    source, payload, payload_cid, and payload_byte_length are under enum.